### PR TITLE
mbedtls: ramdom can use havege if enabled.

### PR DESCRIPTION
**Pull request by p1ng0o**
I didn't succeed to write a "proper" CTR-DRBG random generator. So let's begin with havege 👍 

Anyway, should I save mbedtls_havege_state somewhere ?

### CODE REVIEW

File: lib/vtls/mbedtls.c [hunk 1]
```diff
@@ -729,6 +729,44 @@ size_t Curl_mbedtls_version(char *buffer, size_t size)
...
+
+  ret = mbedtls_ctr_drbg_seed(&ctr_drbg, mbedtls_entropy_func,
+                                &ctr_entropy, NULL, 0);
+
+  if(ret) {
+#ifdef MBEDTLS_ERROR_C
+    mbedtls_strerror(ret, errorbuf, sizeof(errorbuf));
+#endif /* MBEDTLS_ERROR_C */
+    failf(data, "Failed - mbedTLS: ctr_drbg_seed returned (-0x%04X) %s\n",
+          -ret, errorbuf);
```
Comment (bagder) [inline lib/vtls/mbedtls.c]: shouldn't it also return an error here?

**Comment by bagder** [comment]
> should I save mbedtls_havege_state somewhere

You tell me! Should it be saved? When the 'data' pointer is provided, it *could* be stored in that struct.

File: lib/vtls/mbedtls.c [hunk 2]
```diff
@@ -729,6 +729,56 @@ size_t Curl_mbedtls_version(char *buffer, size_t size)
                   (version>>16)&0xff, (version>>8)&0xff);
 }
 
+int Curl_mbedtls_random(struct Curl_easy *data, unsigned char *entropy,
+                     size_t length)
```
Comment (bagder) [inline lib/vtls/mbedtls.c]: I think this is better declared to return `CURLcode` rather than int.

**Comment by p1ng0o** [comment]
The question was, do I privilege cpu time (struct on data), or mem (struct on cstack).
But regarding other vtls implementation of random, and for consistency, It should be better on cstack.

**Comment by bagder** [comment]
Yeah, this function is not used a lot nor in any high performance situation so I think that's totally fine.

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**